aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -350,14 +387,28 @@ inline Return Dispatcher::redispatch(const TypedOperatorHandle<Return (Args...)>
       DispatchKeySet(DispatchKeySet::FULL_AFTER, currentDispatchKey),
       args...
     );
-  return callWithDispatchKey<Return, Args...>(op, dispatchKey, args...);
+  // do not use RecordFunction on redispatch
```

Comment:
Hmm is there a reason to break symmetry here? I think we want `redispatch` to be exactly equivalent to `callWithDispatchKey` on the computed key - basically just a convenience method. Introducing this logic change seems like a potential source of hard-to-track-down inconsistencies in recording results. 

If the change is to avoid double-counting some key other than BackendSelect, maybe that key should be added to `shouldRecord`? cc @smessmer maybe you guys have talked about this already...

Reply:
the reason I do it this way here is that I don't want to run RecordFunction and we don't want to pass a parameter into callWithDispatchKey (as we discussed). Since it's two lines of code I just called the kernel itself

Reply:
Yeah, I definitely have no problem with the implementation. My reaction was to the semantics - my default assumption was that we want `redispatch(op, args)` to have the exactly same behavior as the equivalent `callWithDispatchKey(op, key, args)` call - w.r.t. RecordFunction and everything else. 

But the other POV makes sense too - that a natural consequence of the meaning of "redispatch" is: no (re)recording, so they really are different. 

It's not totally clear to me that the suppression shouldn't move into `shouldRecord`, but it's not a discussion we need to have before you land this 😄 thanks for being patient!